Better error formatting

Right now, makam errors look like this:

```
!! Error in file ./razbor.makam, line 174, characters 1-12:
   Term of type ty -> ty -> ty -> prop,
   whereas expected a type of form ty -> ty -> ty -> ty -> prop.
```

But it would be better if they looked like this:

```
error[E0308]: mismatched types
  --> src/format.rs:52:1
   |
51 |   ) -> Option<String> {
   |        -------------- expected `Option<String>` because of return type
52 | /     for ann in annotations {
53 | |         match (ann.range.0, ann.range.1) {
54 | |             (None, None) => continue,
55 | |             (Some(start), Some(end)) if start > end_index => continue,
...  |
71 | |         }
72 | |     }
   | |_____^ expected enum `std::option::Option`, found ()
```

(this is [annotate-snippets](https://github.com/rust-lang/annotate-snippets-rs) generated error)

Regardless of error formatting, it's important to have `<file name>:line:column` references, because they can be followed by a text editor.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -37,9 +37,10 @@ jobs:
             [[ -e skip ]] || opam config exec make
 
       - run: &prepare-npm-package
-          name: Prepare npm package and install it
+          name: Prepare npm package, test and install it
           command: |
             [[ -e skip ]] || opam config exec make prepare-test-npm-package
+            [[ -e skip ]] || (cd npm; opam config exec yarn jest)
             [[ -e skip ]] || npm install makam-$(./scripts/makam-version.sh npm-test-version).tgz
 
       - run: &tests

--- a/Makefile
+++ b/Makefile
@@ -123,7 +123,7 @@ MAKAM_MARKDOWN_FILES=$(foreach file, $(shell git ls-files '*.md'), --file $(file
 JSOO_RUNTIME_LIBS:=+nat.js +toplevel.js
 
 js:
-	$(OCAMLBUILD) -plugin-tag "package(js_of_ocaml.ocamlbuild)" -no-links js/browser.byte
+	$(OCAMLBUILD) -plugin-tag "package(js_of_ocaml-ocamlbuild)" -no-links js/browser.byte
 	js_of_ocaml -I ./ $(MAKAMFILES) $(MAKAM_MARKDOWN_FILES) $(JSOO_RUNTIME_LIBS) js/myruntime.js _build/js/browser.byte -o js/makam.js
 
 md2makam:

--- a/README.md
+++ b/README.md
@@ -110,7 +110,7 @@ configuration and dependencies separate from other OCaml projects you
 might have. To create a local switch, install all dependencies, and set
 up the environment variables you need, do:
 
-- `opam switch create ./`
+- `opam switch create ./ ocaml-base-compiler.4.07.1`
 - `eval $(opam config env)`
 
 Makam also depends on Node.js, which is used for optimized parser

--- a/_oasis
+++ b/_oasis
@@ -1,6 +1,6 @@
 OASISFormat: 0.4
 Name:        makam
-Version:     0.7.33
+Version:     0.7.34
 Synopsis:    The Makam Metalanguage -- a tool for rapid language prototyping
 Authors:     Antonis Stampoulis <antonis.stampoulis@gmail.com>
 Homepage:    http://astampoulis.github.io/

--- a/_oasis
+++ b/_oasis
@@ -10,7 +10,7 @@ Plugins:     DevFiles (0.4)
 BuildTools:  ocamlbuild
 OCamlVersion:          >= 4.04
 AlphaFeatures:         ocamlbuild_more_args
-XOCamlBuildPluginTags: package(js_of_ocaml.ocamlbuild)
+XOCamlBuildPluginTags: package(js_of_ocaml-ocamlbuild)
 XOCamlBuildExtraArgs:  -byte-plugin
 
 SourceRepository master

--- a/_tags
+++ b/_tags
@@ -10,7 +10,7 @@ true: -traverse, debug
 	       package(camlp4), syntax(camlp4o), package(monad-custom)
 
 <js/*>: package(js_of_ocaml)
-<js/*.ml>: package(js_of_ocaml), package(js_of_ocaml.syntax)
+<js/*.ml>: package(js_of_ocaml)
 
 <bootstrap/*.ml>: package(camlp4.quotations.o)
 <utils/toploopTools.ml>: package(compiler-libs.toplevel), package(camlp4.extend), package(camlp4.fulllib), package(camlp4.quotations.o), parsers

--- a/js/browser.ml
+++ b/js/browser.ml
@@ -6,6 +6,7 @@ open Termlangprolog;;
 open Termlangext;;
 open Termlangrefl;;
 open Termlangparse;;
+open Js_of_ocaml;;
 
 let version = Version.version;;
 
@@ -109,7 +110,7 @@ let (process_input : string -> unit) input =
          (Printf.printf "In %s:\n  Last command was not a query.\n%!"
             (last_cmd_span ()); loop (UChannel.flush_to_furthest input))
       | Peg.IncompleteParse(_, s) ->
-        (print_now ("\nParse error at " ^ s ^ ".\n"); restore_debug (); loop (UChannel.flush_to_furthest input))
+        (print_now ("\nParse error at " ^ (UChannel.string_of_loc s) ^ ".\n"); restore_debug (); loop (UChannel.flush_to_furthest input))
       | e ->
 	raise e
   in

--- a/js/index.html
+++ b/js/index.html
@@ -10,7 +10,7 @@
 
       var worker = new Worker('makam.js');
       var terminal = null;
-      const version = "0.7.33";
+      const version = "0.7.34";
      
       $(function() {
 

--- a/npm/package.json
+++ b/npm/package.json
@@ -1,6 +1,6 @@
 {
   "name": "makam",
-  "version": "0.7.33",
+  "version": "0.7.34",
   "description": "The Makam metalanguage -- a tool for rapid language prototyping",
   "main": "lib/index.js",
   "scripts": {

--- a/npm/src/helpers.js
+++ b/npm/src/helpers.js
@@ -17,9 +17,9 @@ export type MakamResult = {
 };
 
 const parseLocation = (message: string): ?Location => {
-  const sameLineRegexp = /line ([0-9]+), characters ([0-9]+)-([0-9]+)/m;
-  const spansLineRegexp = /line ([0-9]+), character ([0-9]+) to line ([0-9]+), character ([0-9]+)/m;
-  const singleCharRegexp = /line ([0-9]+), character ([0-9]+)/m;
+  const sameLineRegexp = /([0-9]+):([0-9]+)-([0-9]+)/m;
+  const spansLineRegexp = /([0-9]+)\.([0-9]+)-([0-9]+)\.([0-9]+)/m;
+  const singleCharRegexp = /([0-9]+):([0-9]+)/m;
 
   const singleMatch = message.match(singleCharRegexp);
   const sameMatch = message.match(sameLineRegexp);
@@ -49,7 +49,7 @@ const parseLocation = (message: string): ?Location => {
 };
 
 const parseError = (error: string): LocatedMessage => {
-  const errorRegexp = /^!! Error in block block[0-9]*, .*:$/m;
+  const errorRegexp = /^!! Error in block block[0-9]*:.*:$/m;
   return {
     message: error.replace(errorRegexp, "").trim(),
     location: parseLocation(error)
@@ -57,7 +57,7 @@ const parseError = (error: string): LocatedMessage => {
 };
 
 const parseQueryResult = (queryResult: string): LocatedMessage => {
-  const headerRegexp = /^line .+:$/m;
+  const headerRegexp = /^.+:$/m;
   return {
     message: queryResult.replace(headerRegexp, "").trim(),
     location: parseLocation(queryResult)
@@ -67,7 +67,7 @@ const parseQueryResult = (queryResult: string): LocatedMessage => {
 const parseOutput = (output: string): MakamResult => {
   let processed = output;
 
-  const errorRegexp = /^!! Error in block block[0-9]*, .*$\n(.+\n)*\n/gm;
+  const errorRegexp = /^!! Error in block block[0-9]*:.*$\n(.+\n)*\n/gm;
   let errors = [];
   const errorMatch = processed.match(errorRegexp);
   if (errorMatch) {
@@ -75,7 +75,7 @@ const parseOutput = (output: string): MakamResult => {
   }
 
   processed = processed.replace(errorRegexp, "");
-  const queryResultRegexp = /^-- Query result in block block[0-9]*, /m;
+  const queryResultRegexp = /^-- Query result in block block[0-9]*:/m;
   let queryResults = processed.split(queryResultRegexp);
   if (queryResults.length >= 1) {
     queryResults = queryResults.slice(1).map(parseQueryResult);

--- a/npm/src/helpers.test.js
+++ b/npm/src/helpers.test.js
@@ -2,14 +2,14 @@ const { parseLocation } = require("./helpers");
 
 describe("parseLocation", () => {
   it("parses location with a single character", () => {
-    expect(parseLocation("line 5, character 5")).toEqual({
+    expect(parseLocation("5:5")).toEqual({
       start: { line: 3, char: 4 },
       end: { line: 3, char: 4 }
     });
   });
 
   it("parses location in the same line", () => {
-    expect(parseLocation("line 5, characters 5-10")).toEqual({
+    expect(parseLocation("5:5-10")).toEqual({
       start: { line: 3, char: 4 },
       end: { line: 3, char: 9 }
     });
@@ -17,7 +17,7 @@ describe("parseLocation", () => {
 
   it("parses locations spanning different lines", () => {
     expect(
-      parseLocation("line 5, character 5 to line 6, character 10")
+      parseLocation("5.5-6.10")
     ).toEqual({
       start: { line: 3, char: 4 },
       end: { line: 4, char: 9 }

--- a/opam/opam
+++ b/opam/opam
@@ -1,6 +1,6 @@
 opam-version: "2.0"
 name: "makam"
-version: "0.7.33"
+version: "0.7.34"
 maintainer: "Antonis Stampoulis <antonis.stampoulis@gmail.com>"
 authors: [ "Antonis Stampoulis <antonis.stampoulis@gmail.com>" ]
 license: "GPL-3"

--- a/opam/opam
+++ b/opam/opam
@@ -24,7 +24,6 @@ depends: [
   "pa_monad_custom"
   "js_of_ocaml"
   "js_of_ocaml-ocamlbuild"
-  "js_of_ocaml-camlp4"
   "js_of_ocaml-compiler"
   "camlp4"
 ]

--- a/parsing/peg.ml
+++ b/parsing/peg.ml
@@ -2663,7 +2663,7 @@ let string_of_peg g =
   let outstr = IO.output_string ()
   in (PegPrint.printer outstr g; IO.close_out outstr)
   
-exception IncompleteParse of UChannel.t * string
+exception IncompleteParse of UChannel.t * UChannel.loc
   
 let getFullParse result =
   match result with
@@ -2672,7 +2672,7 @@ let getFullParse result =
        | Some _ ->
            raise
              (IncompleteParse (uc,
-                (UChannel.string_of_loc (UChannel.loc uc))))
+                (UChannel.loc uc)))
        | None -> realresult)
   | None -> failwith "failed to parse"
   

--- a/parsing/peg.ml4
+++ b/parsing/peg.ml4
@@ -1102,13 +1102,13 @@ let string_of_peg g =
   PegPrint.printer outstr g;
   IO.close_out outstr ;;
 
-exception IncompleteParse of UChannel.t * string
+exception IncompleteParse of UChannel.t * UChannel.loc
 
 let getFullParse result = 
   match result with
     Some(realresult, uc) -> 
       (match UChannel.get_one uc with
-	  Some _ -> raise (IncompleteParse (uc, UChannel.string_of_loc (UChannel.loc uc)))
+	  Some _ -> raise (IncompleteParse (uc, UChannel.loc uc))
 	| None -> realresult)
   | None -> failwith "failed to parse" ;;
 

--- a/parsing/peg.mli
+++ b/parsing/peg.mli
@@ -149,7 +149,7 @@ val parseGen : pegGrammar -> Camlp4.PreCast.Ast.str_item
   
 val peg_to_file : string -> pegGrammar -> unit
   
-exception IncompleteParse of UChannel.t * string
+exception IncompleteParse of UChannel.t * UChannel.loc
   
 val getFullParse : ('a * UChannel.t) option -> 'a
   

--- a/parsing/peg.mli4
+++ b/parsing/peg.mli4
@@ -103,7 +103,7 @@ module PegPrint :
 val parseGen : pegGrammar -> Camlp4.PreCast.Ast.str_item ;;
 
 val peg_to_file  : string -> pegGrammar -> unit ;;
-exception IncompleteParse of UChannel.t * string ;;
+exception IncompleteParse of UChannel.t * UChannel.loc ;;
 val getFullParse    : ('a * UChannel.t) option -> 'a ;;
 
 

--- a/termlang/termlangcanon.ml
+++ b/termlang/termlangcanon.ml
@@ -1435,7 +1435,7 @@ let typing_handler (type a) (code : unit -> a) : a =
         Printf.sprintf "Term of type %a,\n   whereas expected a type of form %a."
                        Typ.print t1 Typ.print t2
       in
-      Utils.log_error (UChannel.string_of_span e.loc) s; raise TypingError
+      Log.error e.loc s; raise TypingError
     end
 
   | OccursCheck(i, t) ->
@@ -1448,7 +1448,7 @@ let typing_handler (type a) (code : unit -> a) : a =
       let s = Printf.sprintf "Type %a needs to be unified with %a, which contains it."
                  Typ.print tm Typ.print t
       in
-      Utils.log_error (UChannel.string_of_span e.loc) s; raise TypingError
+      Log.error e.loc s; raise TypingError
     end
 
 
@@ -1460,7 +1460,7 @@ let typing_handler (type a) (code : unit -> a) : a =
       let s = Printf.sprintf "Variable %a with type %a does not exist."
                  (ExprU.print ~debug:false) x Typ.print t
       in
-      Utils.log_error (UChannel.string_of_span x.loc) s; raise TypingError
+      Log.error x.loc s; raise TypingError
 
     end
 
@@ -1473,7 +1473,7 @@ let typing_handler (type a) (code : unit -> a) : a =
       let s = Printf.sprintf "Unknown type variable %a."
                  Typ.print t
       in
-      Utils.log_error (UChannel.string_of_span t.loc) s; raise TypingError
+      Log.error t.loc s; raise TypingError
 
     end
 
@@ -1487,7 +1487,7 @@ let typing_handler (type a) (code : unit -> a) : a =
       let s = Printf.sprintf "Type %s used with %d arguments instead of %d as defined."
                  s actual expected
       in
-      Utils.log_error (UChannel.string_of_span t.loc) s; raise TypingError
+      Log.error t.loc s; raise TypingError
 
     end
 
@@ -1500,7 +1500,7 @@ let typing_handler (type a) (code : unit -> a) : a =
       let s = Printf.sprintf "Could not fully determine type; got as far as %a."
                              Typ.print t
       in
-      Utils.log_error (UChannel.string_of_span e.loc) s; raise TypingError
+      Log.error e.loc s; raise TypingError
 
     end
 
@@ -1510,7 +1510,7 @@ let typing_handler (type a) (code : unit -> a) : a =
 
       let t = prepareTypeForUser t in
       let s = Printf.sprintf "The kind %a is invalid." Typ.print t in
-      Utils.log_error (UChannel.string_of_span t.loc) s; raise TypingError
+      Log.error t.loc s; raise TypingError
 
     end
 
@@ -1520,7 +1520,7 @@ let typing_handler (type a) (code : unit -> a) : a =
     begin
 
       let s = Printf.sprintf "Abstraction over type is not allowed here." in
-      Utils.log_error (UChannel.string_of_span t.loc) s; raise TypingError
+      Log.error t.loc s; raise TypingError
 
     end
 
@@ -1529,7 +1529,7 @@ let typing_handler (type a) (code : unit -> a) : a =
     begin
 
       let s = Printf.sprintf "Rank-N-polymorphism not supported." in
-      Utils.log_error (UChannel.string_of_span t.loc) s; raise TypingError
+      Log.error t.loc s; raise TypingError
 
     end
 

--- a/termlang/termlangprolog.ml
+++ b/termlang/termlangprolog.ml
@@ -3396,7 +3396,7 @@ let queryGoal ?(print = false) (goal : exprU) : (string * pattcanon) list RunCtx
     _ <-- mapM (uncurry nameUnify) nu ;
     let _ =
       if print && not !is_interactive then
-        Utils.log_info_header (UChannel.string_of_span goal.loc) "Query result"
+        Log.info_header goal.loc "Query result"
     in
 
     _     <-- ifte
@@ -3466,26 +3466,26 @@ let prolog_handler e =
   try Lazy.force e
   with
   | UnknownPredicate(p) -> begin
-    Utils.log_error (UChannel.string_of_span p.loc)
+    Log.error p.loc
     (Printf.sprintf "The predicate in %a is not concrete."
                     Pattneut.alphaSanitizedPrint p);
     raise PrologError
     end
   | MalformedClause(p) -> begin
-    Utils.log_error (UChannel.string_of_span p.loc)
+    Log.error p.loc
     (Printf.sprintf "The clause %a is malformed."
                     Pattneut.alphaSanitizedPrint p);
     raise PrologError
   end
   | MalformedClauseTypecase(p,t,t') -> begin
-    Utils.log_error (UChannel.string_of_span p.loc)
+    Log.error p.loc
     (Printf.sprintf "The clause conclusion %a specializes parametric type variables.\n   Expected type: %a\n   Actual type: %a"
       Pattneut.alphaSanitizedPrint p
       Typ.print t Typ.print t');
     raise PrologError
   end
   | ClauseForBuiltin(p,s) -> begin
-    Utils.log_error (UChannel.string_of_span p.loc)
+    Log.error p.loc
     (Printf.sprintf "Cannot add new clause to builtin predicate %s." s);
     raise PrologError
   end

--- a/toploop/repl.ml
+++ b/toploop/repl.ml
@@ -132,40 +132,40 @@ let restore_state filename =
 exception ErrorInFile;;
 
 let exception_handler (type a) (f: unit -> a) (recover: unit -> a) (recover_parse: unit -> a) =
-  let last_cmd_span () = UChannel.string_of_span !MakamGrammar.last_command_span in
+  let last_cmd_span () = !MakamGrammar.last_command_span in
   try f ()
   with
   | Termlangcanon.FileNotFound(s, all) ->
-     (Utils.log_error (last_cmd_span ())
+     (Log.error (last_cmd_span ())
      (Printf.sprintf "File %s not found (searched: %a)."
                     s (List.print ~first:"[" ~last:"]" ~sep:"; " String.print) all)
      ; recover())
   | Termlangcanon.TypingError | Termlangprolog.PrologError ->
      (recover())
   | Termlangrefl.StagingError(code) ->
-     (Utils.log_error (UChannel.string_of_span code.loc)
+     (Log.error code.loc
      "Error in staged code."
      ; recover())
   | Termlangprolog.ResetInModule m ->
-     (Utils.log_error (last_cmd_span ())
+     (Log.error (last_cmd_span ())
      (Printf.sprintf "Module %s tried to reset the state." m)
      ; recover())
   | Termlangcanon.NotInModule ->
-     (Utils.log_error (last_cmd_span ())
+     (Log.error (last_cmd_span ())
      "Stopping extension to module, but no module is open."
      ; recover())
   | MakamGrammar.NoTestSuite ->
-     (Utils.log_error (last_cmd_span ())
+     (Log.error (last_cmd_span ())
      "Test suite has not been specified, use %testsuite directive."
      ; recover())
   | MakamGrammar.NoQueryToTest ->
-     (Utils.log_error (last_cmd_span ())
+     (Log.error (last_cmd_span ())
      "Last command was not a query."
      ; recover())
   | MakamGrammar.Forget(s) ->
      (forget_to_state s; recover())
   | Peg.IncompleteParse(_, s) ->
-     (Utils.log_error s
+     (Log.error (UChannel.mk_span s s)
      "Parse error."
      ; recover_parse())
   | ErrorInFile -> recover ()
@@ -189,7 +189,7 @@ let _ =
         let res = prevparser syntax memo mode input in
         match res, UChannel.reached_eof input with
           Some(_, uc), false ->
-            raise (Peg.IncompleteParse(uc, (UChannel.string_of_loc (UChannel.loc uc))))
+            raise (Peg.IncompleteParse(uc, UChannel.loc uc))
         | _ -> restore_interactivity (); res)
         (fun _ -> restore_interactivity (); raise ErrorInFile)
         (fun _ -> restore_interactivity (); raise ErrorInFile))
@@ -250,7 +250,7 @@ let rec repl ?input () : unit =
           | Some(_, input') ->
              store_state (); loop input'
           | None ->
-             raise (Peg.IncompleteParse(input, (input |> UChannel.loc |> UChannel.string_of_loc)))
+             raise (Peg.IncompleteParse(input, input |> UChannel.loc))
       end)
       (fun () -> restore_debug (); if !is_interactive then recover () else (raise ErrorInFile))
       (fun () -> restore_debug (); if !is_interactive then skip_line_and_recover () else (raise ErrorInFile))

--- a/toploop/version.ml
+++ b/toploop/version.ml
@@ -1,2 +1,2 @@
 let version = "0.7.34" ;;
-let source_hash = "258e70cc8a968ae7581c7e6786f716c61c79d496";;
+let source_hash = "182ba6890bfccb4bcfe88bec602d9f82fe483ece";;

--- a/toploop/version.ml
+++ b/toploop/version.ml
@@ -1,2 +1,2 @@
-let version = "0.7.33" ;;
-let source_hash = "4254a434b58a53b81a49e4afb39f4843a1980a5b";;
+let version = "0.7.34" ;;
+let source_hash = "258e70cc8a968ae7581c7e6786f716c61c79d496";;

--- a/utils/log.ml
+++ b/utils/log.ml
@@ -1,0 +1,14 @@
+let error loc s =
+  Printf.printf "\n!! Error in %s:\n   %s\n\n%!" (UChannel.string_of_span loc) s
+;;
+
+let info_header loc type_s =
+  Printf.printf "\n-- %s in %s:\n" type_s (UChannel.string_of_span loc)
+;;
+
+let info loc type_s s =
+  if s = "" then
+    (Printf.printf "\n-- %s in %s.\n\n%!" type_s (UChannel.string_of_span loc))
+  else
+    (Printf.printf "\n-- %s in %s:\n%s\n\n%!" type_s (UChannel.string_of_span loc) s)
+;;

--- a/utils/uChannel.ml
+++ b/utils/uChannel.ml
@@ -6,17 +6,21 @@ type realspan = { startloc : loc ; endloc : loc } ;;
 type span     = realspan option ;;
 
 
-let string_of_loc loc  = Printf.sprintf "%s, line %d, character %d%!" loc.description loc.lineno loc.charno ;;
+let string_of_loc loc  = Printf.sprintf "%s:%d:%d%!" loc.description loc.lineno loc.charno ;;
 let string_of_span span : string =
   match span with
       None -> "<none>"
     | Some({startloc = locstart ; endloc = locend }) ->
       begin
         assert(locstart.description = locend.description);
-        if locstart.lineno = locend.lineno then
-          Printf.sprintf "%s, line %d, characters %d-%d%!" locstart.description locstart.lineno locstart.charno locend.charno
+        if locstart.lineno = locend.lineno then begin
+          if locstart.charno = locend.charno then
+            string_of_loc locstart
+          else
+            Printf.sprintf "%s:%d:%d-%d%!" locstart.description locstart.lineno locstart.charno locend.charno
+          end
         else
-          Printf.sprintf "%s, line %d, character %d to line %d, character %d%!" locstart.description locstart.lineno locstart.charno locend.lineno locend.charno
+          Printf.sprintf "%s:%d.%d-%d.%d%!" locstart.description locstart.lineno locstart.charno locend.lineno locend.charno
       end
 ;;
 

--- a/utils/utils.ml
+++ b/utils/utils.ml
@@ -306,21 +306,6 @@ module ExtList =
 let rec repeatmany n (f : unit -> 'a) = if n = 1 then f () else (let _ = f () in repeatmany (n-1) f);;
 
 
-let log_error loc s =
-  Printf.printf "\n!! Error in %s:\n   %s\n\n%!" loc s
-;;
-
-let log_info_header loc type_s =
-  Printf.printf "\n-- %s in %s:\n" type_s loc
-;;
-
-let log_info loc type_s s =
-  if s = "" then
-    (Printf.printf "\n-- %s in %s.\n\n%!" type_s loc)
-  else
-    (Printf.printf "\n-- %s in %s:\n%s\n\n%!" type_s loc s)
-;;
-
 let time_eval e =
   let start_time = Sys.time () in
   let _ = Lazy.force e in
@@ -335,7 +320,6 @@ let exn_to_option e =
   try
     Some(Lazy.force e)
   with _ -> None
-
 
 let uncurry3 f (a,b,c) = f a b c ;;
 

--- a/webui/package.json
+++ b/webui/package.json
@@ -1,6 +1,6 @@
 {
   "name": "makam-webui",
-  "version": "0.7.33",
+  "version": "0.7.34",
   "description": "CodeMirror-based WebUI for Makam",
   "main": "makam-webui-bundle.js",
   "scripts": {

